endpoints/openrtb2/auction.go

```diff
@@ -1510,12 +1510,12 @@ func (deps *endpointDeps) validateImpExt(imp *openrtb_ext.ImpWrapper, aliases ma
 	errL := []error{}
 
 	for bidder, ext := range prebid.Bidder {
-		coreBidder := bidder
+		coreBidder, _ := openrtb_ext.NormalizeBidderName(bidder)
 		if tmp, isAlias := aliases[bidder]; isAlias {
-			coreBidder = tmp
+			coreBidder = openrtb_ext.BidderName(tmp)
```

Comment:
Oh, I see `validateAliases()` is trying to update the alias dictionary. You should test that `validateAliases()` is successfully updating the alias dictionary, since the dictionary is being passed as a value rather than a reference. A dictionary is a complex object, so it is likely that Go is copying just the reference to the complex object rather than duplicating the whole dictionary, but I don't think this behavior is well defined so we should test that Go behaves as we think if we are going to rely on undefined behavior.

Reply:
thanks Hans! 
This is actually being tested implicitly by testing `validateRequest` method that is the caller of `validateAliases` method. Here is [the test case](https://github.com/prebid/prebid-server/blob/3b753adb20068302acd49728fd2577b7b06da700/endpoints/openrtb2/auction_test.go#L1697-L1727) that covers it and where `validateAliases` is being called upon by testing `validateRequest` method since the bidRequest defines `request.ext.prebid.aliases` and no error is being returned.   

However, I added a separate unit test for validateAliases method here - https://github.com/prebid/prebid-server/pull/3113/commits/98fd465a5d83eae85dcb34deb07df1509549eb59

Reply:
The implicit test does not cover the alias being updated, as all the bidder names are already lower case. But the new explicit test does verify that the alias map is updated with lower case bidders, thank you.

Reply:
ok. To test alias bidder names as lower case, I have added [all-ext-case-insensitive.json](https://github.com/prebid/prebid-server/pull/3113/files#diff-5fa129b5a23471cd8a1ad44b6913e04ec54730fd7eafa94a6e4d1ee97714eac4) that covers the aliases as being case insensitive. Marking this conversation as closed.